src/components/executive-education-2u/ExecutiveEducation2UError.jsx

```diff
@@ -1,97 +1,65 @@
+    description: 'Error message when no learner credit is available to cover this course.',
+  },
+  noOfferWithRemainingApplications: {
+    id: 'ExecutiveEducation2UError.noOfferWithRemainingApplications',
+    defaultMessage: 'Your enrollment was not completed! You have reached your maximum number of allowed enrollments.',
+    description: 'Error message when no learner credit is available to cover this course.',
+  },
+  systemError: {
+    id: 'ExecutiveEducation2UError.systemError',
+    defaultMessage: 'System Error has occurred.',
+    description: 'Error message when no learner credit is available to cover this course.',
+  },
+  default: {
+    id: 'ExecutiveEducation2UError.default',
+    defaultMessage: 'An error has occurred.',
+    description: 'Error message when no learner credit is available to cover this course.',
+  },
+});
 
-  const createExecutiveEducationFailureMessage = (failureCode) => {
-    const failureCodeMessages = {
-      no_offer_available: 'No offer is available to cover this course.',
-      no_offer_with_enough_balance: 'You don’t have access to this course because your organization '
-                                      + 'doesn’t have enough funds. Please contact your edX administrator '
-                                      + 'to resolve the error and provide you access to this content.',
-      no_offer_with_enough_user_balance: 'Your enrollment was not completed! You have already spent your personal budget for enrollments.',
-      no_offer_with_remaining_applications: 'Your enrollment was not completed! You have reached your maximum number of allowed enrollments.',
-      system_error: 'System Error has occurred.',
-      default: 'An error has occurred.',
-    };
-    return failureCodeMessages[failureCode] ? failureCodeMessages[failureCode] : failureCodeMessages.default;
-  };
-  const showHelpfulLink = (failureCode) => {
-    const failureCodeShowLink = {
-      no_offer_available: true,
-      no_offer_with_enough_balance: true,
-      no_offer_with_enough_user_balance: true,
-      no_offer_with_remaining_applications: true,
-      system_error: false,
-      default: false,
-    };
-    return failureCodeShowLink[failureCode] ? failureCodeShowLink[failureCode] : failureCodeShowLink.default;
+export const createExecutiveEducationFailureMessage = ({ failureCode, intl }) => {
+  const failureCodeMessages = {
+    no_offer_available: intl.formatMessage(messages.noOfferAvailable),
+    no_offer_with_enough_balance: intl.formatMessage(messages.noOfferWithEnoughBalance),
+    no_offer_with_enough_user_balance: intl.formatMessage(messages.noOfferWithEnoughUserBalance),
+    no_offer_with_remaining_applications: intl.formatMessage(messages.noOfferWithRemainingApplications),
+    system_error: intl.formatMessage(messages.systemError),
+    default: intl.formatMessage(messages.default),
   };
+  return failureCodeMessages[failureCode] ? failureCodeMessages[failureCode] : failureCodeMessages.default;
```

Comment:
nit: possible shortening
```suggestion
  return failureCodeMessages[failureCode] ?? failureCodeMessages.default;
```